ProjectProtocol/project-protocol-web
# [PRO-296] Resource likes/dislikes UI

Changes
---
* Add resource like/dislike API client functions
* Add ResourceVoteControls component and test, for showing and updating like/dislike counts.
* Show toast when user is not authenticated
* Truncate resource description if more than 100 characters. TODO: translate Show more/less and get design feedback.

Testing:
---
* Try liking/disliking when not signed in
* Try liking/disliking when signed in
* Check that counts update and make sense.

📸 Screenshots
---
Outline icons when user hasn't voted on a resource:
<img width="556" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/e8c6b751-d4ee-488a-98f0-b9d1d0056645">

Filled in icons when user has voted:
<img width="560" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/37308dfd-742c-481e-a54a-b2c42337cf66">

Alert when not signed in:
<img width="425" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/23d1cb83-cc26-45dc-a660-048e62ae767b">
en Español:
<img width="330" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/0ca07107-5391-41c1-8683-aaf914a9fbbe">
